src/api/Metadata.js

```diff
@@ -153,27 +153,220 @@ class Metadata extends File {
+     * @return {Array} metadata instances without classification
+     */
+    extractClassification(id: string, instances: Array<MetadataInstance>): Array<MetadataInstance> {
+        const index = instances.findIndex(instance => instance.$template === METADATA_TEMPLATE_CLASSIFICATION);
+        if (index !== -1) {
+            const classification = instances.splice(index, 1);
+            const cache: APICache = this.getCache();
+            const key = this.getClassificationCacheKey(id);
+            cache.set(key, classification[0]);
+        }
+        return instances;
+    }
+
+    /**
+     * Finds template for a given metadata instance.
+     *
+     * @param {string} id - box file id
+     * @param {Object} instance - metadata instance
+     * @param {Array} templates - metadata templates
+     * @return {Object|undefined} template for metadata instance
+     */
+    async getTemplateForInstance(
+        id: string,
+        instance: MetadataInstance,
+        templates: Array<MetadataEditorTemplate>,
+    ): Promise<?MetadataEditorTemplate> {
+        const instanceId = instance.$id;
+        const templateKey = instance.$template;
+        const scope = instance.$scope;
+        let template = templates.find(t => t.templateKey === templateKey && t.scope === scope);
+        if (scope === METADATA_SCOPE_ENTERPRISE && !template) {
+            // If the template does not exist, it can be a template from another
+            // enterprise because the user is viewing a collaborated file.
+            const crossEnterpriseTemplate = await this.getTemplates(id, instanceId, true);
+            // The API always returns an array of at most one item
+            template = crossEnterpriseTemplate[0]; // eslint-disable-line
+        }
+        return template;
+    }
+
+    /**
+     * Creates and returns metadata editors.
+     *
+     * @param {string} id - box file id
+     * @param {Array} instances - metadata instances
+     * @param {Object} customPropertiesTemplate - custom properties template
+     * @param {Array} enterpriseTemplates - enterprise templates
+     * @param {Array} globalTemplates - global templates
+     * @param {boolean} canEdit - metadata editability
+     * @return {Array} metadata editors
+     */
+    async getEditors(
+        id: string,
+        instances: Array<MetadataInstance>,
+        customPropertiesTemplate: MetadataEditorTemplate,
+        enterpriseTemplates: Array<MetadataEditorTemplate>,
+        globalTemplates: Array<MetadataEditorTemplate>,
+        canEdit: boolean,
+    ): Promise<Array<MetadataEditor>> {
+        // All usable templates for metadata instances
+        const templates: Array<MetadataEditorTemplate> = [customPropertiesTemplate].concat(
+            enterpriseTemplates,
+            globalTemplates,
+        );
+
+        // Filter out skills and classification
+        // let filteredInstances = this.extractSkills(id, instances);
```

Comment:
Still needed?

Reply:
Left it for now as a hint of where to make this upcoming change